### PR TITLE
[BUGFIX beta] Add blueprints for "ember-cli-mocha >= 0.12.0"

### PR DESCRIPTION
**Pull request by Turbo87**
Resolves https://github.com/emberjs/data/issues/4689

### CODE REVIEW

**Comment by Turbo87** [comment]
discovered a small issue with this, do not merge for now... 😞

File: blueprints/test-framework-detector.js [hunk 1]
```diff
@@ -12,7 +14,12 @@ module.exports = function(blueprint) {
     if ('ember-cli-qunit' in dependencies) {
       type = 'qunit';
     } else if ('ember-cli-mocha' in dependencies) {
-      type = 'mocha';
+      var checker = new VersionChecker({ project: this.project });
+      if (fs.existsSync(this.path + '/mocha-0.12-files') && checker.for('ember-cli-mocha', 'npm').satisfies('>=0.12.0')) {
```
Comment (pangratz) [inline blueprints/test-framework-detector.js]: Why is the check for `/mocha-0.12-files` necessary`? Shouldn't this be the case when the addon of this version is installed?

Comment (Turbo87) [inline blueprints/test-framework-detector.js]: @pangratz that check allows us to have some blueprints that care about the Mocha version and others that don't. This is not technically needed in this case, but I've reused the code from https://github.com/emberjs/ember.js/pull/14670 to reduce differences.

**Comment by pangratz** [comment]
🚀 🎉 thanks a bunch @Turbo87

**Pull request merged**